src/optim/FishLeg/fishleg.py

```diff
@@ -278,34 +283,99 @@ def update_aux(self) -> None:
                 u2 = u2 + torch.sum(params.grad * params.grad)
         u_norm = torch.sqrt(u2)
 
-        reg_term = 0.0
-        quad_term = 0.0
-        linear_term = 0.0
+        aux_loss = 0
 
-        p_idx = 0
-        for module in self.model.modules():
-            if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
-                layer_grads = []
-                for name, param in module.named_parameters():
-                    if "fishleg_aux" not in name:
-                        layer_grads.append(param.grad.data / u_norm)
+        if self.method == "antithetic":
+            nat_grads = []
+            p_idx = 0
+            for module in self.model.modules():
+                if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
```

Comment:
would be nicer here (and in the following occurrences) if we maybe just check that it is type FishModule? what do you think?